src/wasm/wasm-binary.cpp

```diff
@@ -2837,8 +2837,9 @@ BinaryConsts::ASTNodes WasmBinaryBuilder::readExpression(Expression*& curr) {
           startPos - codeSectionLocation;
       }
       break;
-    case BinaryConsts::Catch:
-    case BinaryConsts::CatchAll: {
+    case BinaryConsts::Catch: {
+      // TODO: add CatchAll, but the value of the constant is identical to Else
+      // case BinaryConsts::CatchAll: {
```

Comment:
Maybe we can add `CatchAll` handling in `case BinaryConsts::Else:`? We will know if that is not an `Else` but a `CatchAll` if we encountered a `try` but not an `end` yet. But because control flow sturctures can be nested, we would need a stack to track this, something like I used as `procesStack` in #3494...?

Reply:
Thanks, yeah, a stack seems unavoidable. Looks like we already have a control flow stack here, luckily, which I was able to use, PTAL.